src/common/perf_counters_cache.h

```diff
@@ -0,0 +1,234 @@
+#pragma once
+
+#include "common/perf_counters.h"
+#include "common/perf_counters_key.h"
+#include "common/ceph_context.h"
+#include "common/ceph_mutex.h"
+#include "common/intrusive_lru.h"
+
+template <typename LRUItem>
+struct item_to_key {
+  using type = std::string;
+  const type &operator()(const LRUItem &item) {
+    return item.labels;
+  }
+};
+
+struct PerfCountersCacheEntry : public ceph::common::intrusive_lru_base<
+                                       ceph::common::intrusive_lru_config<
+                                       std::string, PerfCountersCacheEntry, item_to_key<PerfCountersCacheEntry>>> {
+  std::string labels;
+  std::shared_ptr<PerfCounters> counters;
+  CephContext *cct;
+
+  PerfCountersCacheEntry(std::string _key) : labels(_key) {}
```

Comment:
functions should either take `std::string` by const ref, or move the argument like `: labels(std::move(_key))` to avoid copying the string twice